lib/net.js

```diff
@@ -1131,6 +1205,20 @@ function lookupAndConnect(self, options) {
   }
   port |= 0;
 
+  if (typeof autoSelectFamily !== 'undefined') {
```

Comment:
```suggestion
  if (autoSelectFamily !== undefined) {
```
or even, if we want to accept `null` as meaning the default value
```suggestion
  if (autoSelectFamily != null) {
```